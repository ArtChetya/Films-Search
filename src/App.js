import React from 'react';
import { hot } from 'react-hot-loader/root';
import 'typeface-roboto';
// import { HomeContainer } from './pages/Home/HomeContainer';
import { FilmDetailsPageContainer } from './pages/FilmDetailsPage/FilmDetailsPageContainer';

function App() {
    return (
        <>
            <FilmDetailsPageContainer />
        </>
    );
}

export default hot(App);
