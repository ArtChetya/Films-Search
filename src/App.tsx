import { ErrorBoundary } from 'components';
import * as React from 'react';
import { hot } from 'react-hot-loader/root';
import { Provider } from 'react-redux';
import { StaticRouterContext } from 'react-router';
import { Redirect, Route, Switch } from 'react-router-dom';
import { filmDetailsPageConnector } from './pages/FilmDetailsPage/FilmDetailsPageConnector';
import { FilmDetailsPageContainer } from './pages/FilmDetailsPage/FilmDetailsPageContainer';
import { homeConnector } from './pages/Home/HomeConnector';
import { HomeContainer } from './pages/Home/HomeContainer';
import { NotFound } from './pages/NotFound';

const HomeContainerConnected = homeConnector(HomeContainer);
const filmDetailsPageContainerConnected = filmDetailsPageConnector(
    FilmDetailsPageContainer
);

// App.propTypes = {
//     Router: PropTypes.func.isRequired,
//     location: PropTypes.string,
//     context: PropTypes.shape({
//         url: PropTypes.string
//     }),
//     store: PropTypes.shape({
//         dispatch: PropTypes.func.isRequired,
//         getState: PropTypes.func.isRequired
//     }).isRequired
// };
// App.defaultProps = {
//     location: null,
//     context: null
// };

interface IAppProps {
    Router: React.ElementType;
    location?: string | object;
    context?: StaticRouterContext;
    store: any;
}

const App: React.FunctionComponent<IAppProps> = ({
    Router,
    location,
    context,
    store
}) => {
    return (
        <Provider store={store}>
            <ErrorBoundary>
                <Router location={location} context={context}>
                    <Switch>
                        <Route
                            exact={true}
                            path="/"
                            component={HomeContainerConnected}
                        />
                        <Redirect exact={true} from="/search" to="/" />
                        <Route
                            path="/movie/:id"
                            component={filmDetailsPageContainerConnected}
                        />
                        <Route
                            path="/search/:query"
                            component={HomeContainerConnected}
                        />
                        <Route path="/not-found" component={NotFound} />
                        <Redirect to="/not-found" />
                    </Switch>
                </Router>
            </ErrorBoundary>
        </Provider>
    );
};

export default hot(App);
