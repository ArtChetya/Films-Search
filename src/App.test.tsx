import * as enzyme from 'enzyme';
import * as React from 'react';
import App from './App';

describe('App component', () => {
    it('should be rendered correctly', () => {
        const Router = () => <div>Router</div>;
        const location = 'some-location';
        const context = { url: 'some-url' };
        const store = {
            dispatch: jest.fn(),
            getState: jest.fn(),
            subscribe: jest.fn()
        };

        const component = enzyme.shallow(
            <App
                Router={Router}
                location={location}
                context={context}
                store={store}
            />
        );
        expect(component).toMatchSnapshot();
    });
});
