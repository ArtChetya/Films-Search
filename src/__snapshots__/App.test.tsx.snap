// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should be rendered correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
      "subscribe": [MockFunction],
    }
  }
>
  <ErrorBoundary>
    <Router
      context={
        Object {
          "url": "some-url",
        }
      }
      location="some-location"
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Redirect
          exact={true}
          from="/search"
          push={false}
          to="/"
        />
        <Route
          component={[Function]}
          path="/movie/:id"
        />
        <Route
          component={[Function]}
          path="/search/:query"
        />
        <Route
          component={[Function]}
          path="/not-found"
        />
        <Redirect
          push={false}
          to="/not-found"
        />
      </Switch>
    </Router>
  </ErrorBoundary>
</Provider>
`;
